src/crystal/init_cli.cr

```diff
@@ -0,0 +1,141 @@
+require "ecr/macros"
+require "option_parser"
+
+module Crystal
+  module InitCli
+    def self.run(args)
+      options = {} of Symbol => String
+
+      OptionParser.parse(args) do |opts|
+        opts.banner = %{USAGE: crystal init TYPE NAME [DIR]
+
+TYPE is one of:
+    lib                      creates library skeleton
+    app                      creates application skeleton
+
+NAME - name of project to be generated,
+       eg: example
+DIR  - directory where project will be generated,
+       default: NAME, eg: ./custom/path/example
+}
+
+        opts.on("--help", "Shows this message") do
+          puts opts
+          exit
+        end
+
+        opts.unknown_args do |args, after_dash|
+          options[:skeleton_type] = fetch_skeleton_type(opts, args)
+          options[:name] = fetch_required_parameter(opts, args, "NAME")
+          options[:dir] = args.empty? ? options[:name] : args.shift
+        end
+      end
+
+      options[:author] = `git config --get user.name`.strip
+      InitProject.new(options[:skeleton_type],
+                      options[:name],
+                      options[:dir],
+                      options[:author]).run
+    end
+
+    def self.fetch_skeleton_type(opts, args)
+      skeleton_type = fetch_required_parameter(opts, args, "TYPE")
+      unless ["lib", "app"].includes?(skeleton_type)
```

Comment:
Here you can do:

```crystal
unless {"lib", "app"}.includes?(skeleton_type)
```

This avoids an array allocation :-)

(yes, I know, performance obsession :-P)